src/linux.rs

```diff
@@ -742,6 +731,29 @@ pub fn detect_linux() -> Result<OsVersion, Box<dyn Error>> {
     })
 }
 
+fn determine_id_from_os_release_contents(lines: &[String]) -> Result<String, Box<dyn Error>> {
```

Comment:
I extracted this pretty much as it was just to demonstrate extraction of the PopOS distro name / ID from `/etc/os-release` contents.